internal/webhooks/clusterclass.go

```diff
@@ -220,6 +225,33 @@ func (webhook *ClusterClass) validate(ctx context.Context, oldClusterClass, newC
 	return nil
 }
 
+func validateInfraClusterNamingStrategy(infraClusterNamingStrategy *clusterv1.InfrastructuresNamingStrategy, pathPrefix *field.Path) field.ErrorList {
+	var allErrs field.ErrorList
+
+	if infraClusterNamingStrategy.Template != nil {
+		name, err := topologynames.InfraClusterNameGenerator(*infraClusterNamingStrategy.Template, "cluster", "infraCluster").GenerateName()
+		if err != nil {
+			allErrs = append(allErrs,
+				field.Invalid(
+					pathPrefix.Child("template"),
+					infraClusterNamingStrategy.Template,
+					fmt.Sprintf("invalid template: %v", err),
+				))
+		} else {
+			for _, err := range validation.IsDNS1123Subdomain(name) {
+				allErrs = append(allErrs,
+					field.Invalid(
+						pathPrefix.Child("template"),
+						infraClusterNamingStrategy.Template,
+						fmt.Sprintf("invalid template, generated names would not be valid Kubernetes object names: %v", err),
+					))
+			}
+		}
+	}
+
+	return allErrs
```

Comment:
Let's please move this code to validateNamingStrategies directly above the validation of ControlPlane.NamingStrategy

Reply:
Let's please also sync up the implementation so it's roughly the same as the one for control plane naming strategy (for consistency)